tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs

```diff
@@ -2,467 +2,1586 @@
+                        span = Span<int>.DangerousCreate(testClass, ref testClass.C0[0], testClass.C0.Length);
+        }
 
-        for (int i = 0; i < Iterations; i++)
+        [Benchmark(InnerIterationCount = BaseIterations)]
+        [InlineData(1)]
+        [InlineData(5)]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(500)]
+        [InlineData(1000)]
+        [InlineData(10000)]
+        [InlineData(10000000)]
+        public static void TestSpanDangerousCreateString(int length)
         {
-            Array.Copy(unsortedData, data, Size);
-            TestQuickSortSpan(span);
+            TestClass<string> testClass = new TestClass<string>();
+            testClass.C0 = new string[length];
+            Span<string> span;
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                        span = Span<string>.DangerousCreate(testClass, ref testClass.C0[0], testClass.C0.Length);
         }
-    }
 
-    static void BubbleSortSpanBase()
-    {
-        int[] data = new int[Size];
-        int[] unsortedData = GetUnsortedData();
-        Span<int> span = new Span<int>(data);
+        [Benchmark(InnerIterationCount = BaseIterations)]
+        [InlineData(1)]
+        [InlineData(5)]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(500)]
+        [InlineData(1000)]
+        [InlineData(10000)]
+        [InlineData(10000000)]
+        public static void TestSpanDangerousCreateTestStruct(int length)
+        {
+            TestClass<TestStruct> testClass = new TestClass<TestStruct>();
+            testClass.C0 = new TestStruct[length];
+            Span<TestStruct> span;
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                        span = Span<TestStruct>.DangerousCreate(testClass, ref testClass.C0[0], testClass.C0.Length);
+        }
+        #endregion
 
-        for (int i = 0; i < Iterations; i++)
+        #region TestSpanDangerousGetPinnableReference<T>
+        [Benchmark(InnerIterationCount = BaseIterations)]
+        [InlineData(1)]
+        [InlineData(5)]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(500)]
+        [InlineData(1000)]
+        [InlineData(10000)]
+        [InlineData(10000000)]
+        public static void TestSpanDangerousGetPinnableReferenceByte(int length)
         {
-            Array.Copy(unsortedData, data, Size);
-            TestBubbleSortSpan(span);
+            var array = new byte[length];
+            var span = new Span<byte>(array);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
```

Comment:
The tests that were part of the pre-existing set are tuned that way in this change -- nominal times measured locally are 0.5 - 3 seconds. Not all at 1 second since the bubble and quick sorts are doing the same work and bubble sort is slower.